src/incompletecholesky.jl

```diff
@@ -2,26 +2,36 @@ function assert_pd(d, α)
     @assert α == 0 && all(x -> x > 0, d) "The input matrix is not positive definite."
 end
 
+"""
+    CholeskyPreconditioner(A, memory=2; kwargs...)
+Construct
```

Comment:
We should use JuliaFormatter at some point to have consistent formatting. The lack of an empty line before `Construct` and short docstring lines strike me as odd but it's not a blocker. I am happy to follow any standard automated format.